File: Bai_39/main.cpp
#include<iostream>
#include<vector>
using namespace std;

void beautiful_permutation(int N)
{
    if(N < 1)
    {
        cout << 1 << endl;
        return;
    }
    if(N < 4)
    {
        cout << "NO SOLUTION" << endl;
        return;
    }
    vector<int>permutation;
    for(int i = N; i >= 1; i--)
        if(i % 2 == 0)
            permutation.push_back(i);
    for(int i = N; i >= 1; i--)
        if(i % 2 != 0)
            permutation.push_back(i);
    for(int i : permutation)
        cout << i << ' ';
}

int main()
{
    int N; cin >> N;
    beautiful_permutation(N);
}

File: Bai_39/main.py
def beautiful_permutation(N):
    if N < 1:
        print(1)
        return
    if N < 4:
        print("NO SOLUTION")
        return

    permutation = []
    for i in range(N, 0, -1):
        if i % 2 == 0:
            permutation.append(i)

    for i in range(N, 0, -1):
        if i % 2 != 0:
            permutation.append(i)

    print(*permutation)

N = int(input())
beautiful_permutation(N)
